### PR TITLE
Fix Status view contents overflowing

### LINKED ISSUE #1182: Status View contents overflow narrow containers
# Issue Description
The contents of Status View are currently bleeding out of it's container when it's too narrow for the contents. Worry is that if application supplies long, non-broken Domain, Error strings, or Exception codes as part of the message, they may not wrap within the view correctly as intended. (Is a discovered possible edge-case usage of the component.)

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
1. `Message` and `Title` probably should add an extra css to handle word-breaks and hyphen usage unless there is a performance risk.
```
-ms-word-break: break-all;
    word-break: break-all;

/* Non standard for WebKit */
     word-break: break-word;

-webkit-hyphens: auto;
   -moz-hyphens: auto;
        hyphens: auto;
```

2. Probably should add extra breakpoint to hide the Icon when the container is less than 100px wide (or the appropriately needed space).

## Current Behavior
1. Screenshot showing current behavior of text overflow with a long unbroken "word" in both both the Title and Message
> screen shot 2018-01-29 at 4 43 20 pm

2. Screenshot showing current behavior of icon overflow (and if word wrapping and hyphens were added to the text)
> screen shot 2018-01-29 at 5 10 03 pm

## Steps to Reproduce
1. Text Overflows: Add Status View inside a parent container that has a smaller width than the length needed for a long unbroken word/s used in the title or message. 
2. Icon Overflows: Add Status View inside a parent container that has a smaller width than the icon -- 100px (not ideal and not recommended by UX but could happen). 

## Environment

* Component Version: 1.0.1
* Browser Name and Version: Chrome - Version 63.0.3239.132 (Official Build) (64-bit)
* Operating System and version (desktop or mobile): Mac Sierra version 10.12.5

### PR DESCRIPTION
**Pull request by Blackop778**
### Summary
Prevents long words from extending outside the Status View and will hide the Glyph if it is too wide or too tall to fit in the Status View.

Closes #1182 

### Deployment
* https://terra-core-deployed-pr-1568.herokuapp.com/#/tests/terra-status-view/status-view/status-view-thin-constraint

Thanks for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-status-view/src/terra-dev-site/test/status-view/StatusViewThinConstraint.test.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import Base from 'terra-base';
+import Button from 'terra-button';
+import StatusView from '../../../StatusView';
+
+const locale = document.getElementsByTagName('html')[0].getAttribute('lang');
+const ipsum = 'Lorem ipsum dolor';
+
+export default () =>
+  <Base locale={locale}>
```
Comment (emilyrohrbough) [inline packages/terra-status-view/src/terra-dev-site/test/status-view/StatusViewThinConstraint.test.jsx]: Lets drop the base wrapper. Before terra-dev-site, each test page was standalone from the site which meant they needed to supply their own base wrapper. terra-dev-site now provides this for us! (yay). This is code cleanup thats needs to happen at some point, but haven't quite gotten there.

File: packages/terra-status-view/src/StatusView.scss [hunk 2]
```diff
@@ -12,7 +12,16 @@
...
+    /* stylelint-disable property-no-vendor-prefix */
+    -moz-hyphens: auto;
+    -webkit-hyphens: auto;
+    /* stylelint-enable */
+    hyphens: auto;
     width: 100%;
+    -ms-word-break: break-all;
+    word-break: break-all;
+    /* Non standard for WebKit */
+    word-break: break-word;
```
Comment (emilyrohrbough) [inline packages/terra-status-view/src/StatusView.scss]: Our webpack configuration will handle the vendor-prefixes for us, we these can can be simplified to 
```
hyphens: auto;
word-break: break-word;
```

File: packages/terra-status-view/CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ Unreleased
 ----------
 ### Changed
 * Updated "test:jest" script to work on windows.
+* Fixed text and glyphs being able to overflow out of the Status View
```
Comment (emilyrohrbough) [inline packages/terra-status-view/CHANGELOG.md]: Can you add the heading `### Fixed` above this change? We follow [this guideline](https://keepachangelog.com/en/1.0.0/) to structure our changelogs so they are easier to see what occurred in each release. I just added this link to the doc we sent (I realized it wasn't on there!). 😄

File: packages/terra-status-view/src/StatusView.scss [hunk 4]
```diff
@@ -12,13 +12,8 @@
...
     height: 100%;
-    /* stylelint-disable property-no-vendor-prefix */
-    -moz-hyphens: auto;
-    -webkit-hyphens: auto;
-    /* stylelint-enable */
     hyphens: auto;
     width: 100%;
-    -ms-word-break: break-all;
     word-break: break-all;
     /* Non standard for WebKit */
```
Comment (emilyrohrbough) [inline packages/terra-status-view/src/StatusView.scss]: whoops. remove comment.

File: packages/terra-status-view/src/StatusView.jsx [hunk 5]
```diff
@@ -117,8 +117,8 @@ class StatusView extends React.Component {
       componentsHeight += this.dividerNode.offsetHeight;
     }
 
-    if (this.glyphNode && viewHeight >= componentsHeight + this.glyphNode.offsetHeight) {
-      // if have a glyph to show and can fit inside the containers height, distribute remaining padding to center
+    if (this.glyphNode && viewHeight >= componentsHeight + this.glyphNode.offsetHeight && this.innerNode.offsetWidth >= this.glyphNode.offsetWidth) {
+      // if have a glyph to show and can fit inside the containers height and width, distribute remaining padding to center
```
Comment (yuderekyu) [inline packages/terra-status-view/src/StatusView.jsx]: nit: can we reword the first part of this comment, maybe "if glyph exists and can..."

**Comment by PaulSolDev** [comment]
Functional review complete.  No issues identified.

**Comment by neilpfeiffer** [comment]
Tested: PR #1568
Resolves #1182 
 
**Reviewed:**
- Fixes problems logged in issue.
 
\
Verified by @neilpfeiffer, Ready for Release.
 
**Pass** |
-----------|
UX Review|

**Pull request merged**